packages/teams-js/src/public/location.ts

```diff
@@ -48,25 +48,76 @@ export namespace location {
+  // for use through teamsjs-sdk 2.0. If the user DOES refresh the app after calling this the iframe
+  // would have the new allow parameters, but only the AppPermissions dialog should trigger the
+  // "ask the user to refresh" flow
+  export function requestPermission(): Promise<boolean> {
+    const permissions: DevicePermission[] = [{ type: DevicePermissionType.GeoLocation }];
+
+    return new Promise<boolean>(resolve => {
+      resolve(sendAndHandleError('permission.request', permissions));
+    });
+  }
+
+  // Example of how developer would use permissions functions
+  // export async function permissionExample(): Promise<void> {
+  //   hasPermission().then(alreadyHadPermission => {
+  //     if (!alreadyHadPermission) {
+  //       requestPermission().then(permissionStatus => {
+  //         if (permissionStatus) {
+  //           alert('Use function that required permission');
+  //         } else {
+  //           alert('User was asked to grant permission and refused');
+  //         }
+  //       });
+  //     } else {
+  //       alert('User has previously granted permission, call function that required permission');
+  //     }
+  //   });
+  // }
+
+  export function isSupported(): boolean {
+    return runtime.supports.location ? true : false;
+  }
+
   /**
+   * IMPLEMENTATION NOTE: this should really just be "the unpromisified version of getLocation".
+   * There's no reason to go from callback to promise back to callback in the "real" implementation
    * @deprecated
-   * As of 2.0.0-beta.1, please use {@link location.getLocation location.getLocation(props: LocationProps): Promise\<Location\>} instead.
+   * As of 2.0.0-beta.4, please use one of the following functions:
+   * - {@link location.getCurrentLocation location.getCurrentLocation(): Promise\<Location\>}
+   * - {@link location.map.chooseLocation location.map.chooseLocation(): Promise\<Location\>}
    * @param props {@link LocationProps} - Specifying how the location request is handled
    * @param callback - Callback to invoke when current user location is fetched
    */
-  export function getLocation(props: LocationProps, callback: (error: SdkError, location: Location) => void): void;
-  export function getLocation(
-    props: LocationProps,
-    callback?: (error: SdkError, location: Location) => void,
-  ): Promise<Location> {
+  export function getLocation(props: LocationProps, callback: (error: SdkError, location: Location) => void): void {
     ensureInitialized(FrameContexts.content, FrameContexts.task);
-    return callCallbackWithErrorOrResultFromPromiseAndReturnPromise<Location>(getLocationHelper, callback, props);
+
+    if (!isCurrentSDKVersionAtLeast(locationAPIsRequiredVersion)) {
+      throw { errorCode: ErrorCode.OLD_PLATFORM };
+    }
+    if (!props) {
+      throw { errorCode: ErrorCode.INVALID_ARGUMENTS };
+    }
+
+    callCallbackWithErrorOrResultFromPromiseAndReturnPromise<Location>(getLocationHelper, callback, props);
```

Comment:
I think you want callCallbackWithSDKErrorOrResultFromPromiseAndReturnPromise since you're returning an sdk error but it seems both of those functions take sdkErrors so I could be wrong